lib/ansible/module_utils/basic.py

```diff
@@ -1611,7 +1611,12 @@ def _check_type_float(self, value):
 
     def _check_type_path(self, value):
         value = self._check_type_str(value)
-        return os.path.expanduser(os.path.expandvars(value))
+        valid_path = os.path.expanduser(os.path.expandvars(value))
+
+        if valid_path == '':
```

Comment:
And that 'if' is what im having issues with, I'm not sure it is invalid.

In any case, if it is None it doesn't even hit this function, None is a valid value for paths in many modules (most of the file modules check for this specifically).